### PR TITLE
amp-script: Make inline scripts valid, add max-age

### LINKED ISSUE #24309: amp-script: Local scripts are invalid
Local scripts via `amp-script[script]` attribute are currently invalid pending SxG security review. In the meantime, we recommend using remote scripts via `[src]`.

https://github.com/ampproject/amphtml/issues/24171#issuecomment-524379878

### PR DESCRIPTION
**Pull request by choumx**
Fixes #24309.

- Uncomment inline script validation
- `max-age` only accepts numeric chars and requires `script` attribute

Let me know when this is safe to submit.

/to @twifkak

### CODE REVIEW

File: extensions/amp-script/amp-script.md [hunk 1]
```diff
@@ -192,13 +196,27 @@ Applies extra restrictions to DOM that may be mutated by this `<amp-script>`. Si
 
 - `allow-forms`: Allows [form elements](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/elements) to be created and modified. AMP requires special handling to prevent unauthorized state changing requests from user input. See amp-form's [security considerations](https://amp.dev/documentation/components/amp-form#security-considerations) for more detail.
 
+**max-age (optional)**
+
+Requires the `script` attribute.
+
+The `max-age` attribute specifies the maximum lifetime in seconds the local script is allowed to be served from the time of [signed exchange](https://amp.dev/documentation/guides-and-tutorials/optimize-and-measure/signed-exchange/) publishing.
```
Comment (twifkak) [inline extensions/amp-script/amp-script.md]: I'd mention that when choosing an appropriate max-age, they should evaluate the risk of [downgrades](https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html#seccons-downgrades) of the script.

File: extensions/amp-script/validator-amp-script.protoascii [hunk 2]
```diff
@@ -24,41 +24,47 @@ tags: {
...
+    name: "max-age"
+    value_regex: "[0-9]+"
+    trigger: {
+      # max-age only applies to inline scripts.
+      also_requires_attr: "script"
+    }
+  }
   attrs: { name: "sandbox" }
   attrs: {
     name: "script"
```
Comment (twifkak) [inline extensions/amp-script/validator-amp-script.protoascii]: Consider using the ValueSet feature to validate that script="" points to an existing script tag on the page.
 - Add `AMP_SCRIPT_IDS` to https://github.com/ampproject/amphtml/blob/2424d4f555a102c61e42cf486fcab5f1321a4793/validator/validator.proto#L131.
 - Add `value_one_of_set: AMP_SCRIPT_IDS` here.
 - Copy the id attr spec from https://github.com/ampproject/amphtml/blob/2424d4f555a102c61e42cf486fcab5f1321a4793/validator/validator-main.protoascii#L5378 into the new script tag spec above, and add `add_value_to_set: AMP_SCRIPT_IDS` to it.

File: extensions/amp-script/validator-amp-script.protoascii [hunk 3]
```diff
@@ -24,41 +24,47 @@ tags: {
...
+  spec_url: "https://amp.dev/documentation/components/amp-script"
+}
 tags: {  # <amp-script>
   html_format: AMP
   tag_name: "AMP-SCRIPT"
   disallowed_ancestor: "AMP-SCRIPT"
   requires_extension: "amp-script"
+  attrs: {
+    name: "max-age"
+    value_regex: "[0-9]+"
```
Comment (twifkak) [inline extensions/amp-script/validator-amp-script.protoascii]: Maybe `0|[1-9][0-9]*` to disallow 0-prefix numbers? :man_shrugging:

**Comment by twifkak** [comment]
Also, would like to wait on https://github.com/ampproject/amppackager/pull/347 to merge plus one more PR that integrates that new field.

**Comment by twifkak** [comment]
https://github.com/ampproject/amppackager/pull/348 is merged.

File: extensions/amp-script/validator-amp-script.protoascii (hunk 3 above)
Comment (choumx) [reply extensions/amp-script/validator-amp-script.protoascii]: This is consistent with a lot of other `value_regex` number values though I don't feel strongly either way.

File: extensions/amp-script/amp-script.md (hunk 1 above)
Comment (choumx) [reply extensions/amp-script/amp-script.md]: Took a stab, PTAL.

**Comment by choumx** [comment]
@twifkak Friendly ping. :)

File: validator/validator-main.protoascii [hunk 4]
```diff
@@ -5379,7 +5379,7 @@ attr_lists: {
     name: "id"
     mandatory: true
     # When updating this blacklisted_value_regex, also update the entry for
-    # "id" in $GLOBAL_ATTRS.
+    # "id" in $GLOBAL_ATTRS and "id" in "amp-script extension local script".
```
Comment (twifkak) [inline validator/validator-main.protoascii]: Thanks!

**Pull request merged**